src/main/java/com/google/sps/data/Event.java

```diff
@@ -0,0 +1,199 @@
+
+  // fields to be saved for parsing
+  private String strStartTime;
+  private String strEndTime;
+ 
+  // class constants
+  private static final int HALFHOUR = 30;
+  private static final int HOUR = 60;
+  private static final int MINUTES_IN_A_DAY = 1440;
+  private static final int MIN_POSSIBLE_TIME = 0;
+
+  /**
+   * Constructor that takes in time spent at location
+   * 
+   * @param name name of the location (NOT address)
+   * @param address exact address of the POI
+   * @param date date from user input travel day. Cannot be null.
+   * @param startTime start of activity (HHMM format)
+   * @param travelTime time spent traveling to next location (minutes). 
+   *        Null if last location of the day.
+   * @param timeAtLocation time spent at POI (minutes)
+   */
+  public Event(String name, String address, String date, int startTime, 
+              int travelTime, int timeAtLocation) {
+    this.name = name;
+    this.address = address;
+    this.date = date;
+    this.startTime = startTime;
+    this.travelTime = travelTime;
+    this.endTime = calculateEndTime(timeAtLocation);
+  
+    createCalendarTimes();
+  }
+ 
+  /**
+   * Constructor that assume one hour default spent at location
+   * 
+   * @param name name of the location (NOT address)
+   * @param address exact address of the POI
+   * @param date date from user input travel day. Cannot be null.
+   * @param startTime start of activity (HHMM format)
+   * @param travelTime time spent traveling to next location (minutes)
+   *        Null if last location of the day.
+   */
+  public Event(String name, String address, String date, int startTime, int travelTime) {
+    this(name, address, date, startTime, travelTime, HOUR);
+  }
+
+  // function that sets start and end Time with correct string format
+  private void createCalendarTimes() {
+    try {
+      this.strStartTime = createStrTime(this.date, this.startTime);
+      this.strEndTime = createStrTime(this.date, this.endTime);
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * function that builds string w correct format for calendar-script.js
+   * 
+   * @param date date from user input travel day. Cannot be null.
+   * @param time time to to start or end event. Must be within bounds
+   */
+  private static String createStrTime(String date, int time) throws Exception {
+    if (date == null) {
+      throw new NullPointerException("Date cannot be null");
+    }
+
+    if (time < MIN_POSSIBLE_TIME) {
+      throw new IllegalArgumentException("Time cannot be less than " + MIN_POSSIBLE_TIME);
+    }
+
+    if (time >= MINUTES_IN_A_DAY) {
+      throw new IllegalArgumentException("Time cannot be more than " + MINUTES_IN_A_DAY);
+    }
+
+    String output = "";
+    output += date + "T" + Integer.toString(time).substring(0, 2) + ":" + 
+      Integer.toString(time).substring(2, 4) + ":00";
+
+    try {
+      SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss");
```

Comment:
optional: I think you can construct a SimpleDateFormat which more closely matches the input format and save the trouble of reformatting the string above, eg: `new SimpleDateFormat("yyyy-MM-ddHHmm")`. then pass in `date + Integer.toString(time)`.

Reply:
For the moment if all is working I think I'll keep what I have but I'll keep this in mind going forward!